source/material_model/steinberger.cc

```diff
@@ -551,7 +580,15 @@ namespace aspect
                                   + " fields of type chemical composition."));
 
           has_background_field = (equation_of_state.number_of_lookups() == n_chemical_fields + 1);
-
+          std::vector<std::string> list_of_composition_names = this->introspection().get_composition_names();
```

Comment:
I think you want fields that correspond to chemical compositions, rather than all compositional fields:
```suggestion
          std::vector<std::string> list_of_composition_names = this->introspection().chemical_composition_field_names();
```
This might fix a couple of the tests.